File: scenebuilder/src/main/java/com/sfxcode/sapphire/javafx/control/DualDataListView.java
package com.sfxcode.sapphire.javafx.control;

import com.sfxcode.sapphire.javafx.skin.DataListViewSkin;
import com.sfxcode.sapphire.javafx.skin.DualDataListViewSkin;
import javafx.scene.control.Control;
import javafx.scene.control.Skin;


public class DualDataListView<S> extends Control {


    public DualDataListView() {
        getStyleClass().add("dual-data-list-view");
    }

    protected Skin<DualDataListView<S>> createDefaultSkin() {
        return new DualDataListViewSkin<S>(this);
    }

    @Override public String getUserAgentStylesheet() {
        return getClass().getResource("dualdatalistview.css").toExternalForm();
    }
}

File: src/main/scala/com/sfxcode/sapphire/javafx/control/DualDataListView.scala
package com.sfxcode.sapphire.javafx.control

import javafx.scene.control.{ Control, Skin }

import com.sfxcode.sapphire.javafx.skin.DualDataListViewSkin

class DualDataListView[S <: AnyRef] extends Control {

  lazy val css = getClass.getResource("dualdatalistview.css").toExternalForm

  getStyleClass.add("dual-data-list-view")

  val leftDataListView = new DataListView[S]()

  val rightDataListView = new DataListView[S]()

  protected override def createDefaultSkin: Skin[DualDataListView[S]] =
    new DualDataListViewSkin[S](this)

  override def getUserAgentStylesheet: String = css

  def setItems(left: Iterable[S], right: Iterable[S] = List[S]()): Unit = {
    leftDataListView.setItems(left)
    rightDataListView.setItems(right)
  }

  def setCellProperty(value: String): Unit = {
    leftDataListView.cellProperty.set(value)
    rightDataListView.cellProperty.set(value)
  }

}
